android/src/main/kotlin/com/shopify/reactnative/flash_list/AutoLayoutShadow.kt

```diff
@@ -15,13 +17,21 @@ class AutoLayoutShadow {
     private var lastMaxBound = 0 // Tracks where the last pixel is drawn in the visible window
     private var lastMinBound = 0 // Tracks where first pixel is drawn in the visible window
 
+    private var isInitialRender = true
+    private var anchorStableId = ""
+    private var anchorOffset = 0
+
     /** Checks for overlaps or gaps between adjacent items and then applies a correction (Only Grid layouts with varying spans)
      * Performance: RecyclerListView renders very small number of views and this is not going to trigger multiple layouts on Android side. Not expecting any major perf issue. */
-    fun clearGapsAndOverlaps(sortedItems: Array<CellContainer>) {
+    fun clearGapsAndOverlaps(sortedItems: Array<CellContainer>, scrollView: ScrollView, maintainTopContentPosition: Boolean) {
```

Comment:
I think the flag can be maintained as a variable within this class and final diff can be applied by AutoLayout. That will keep this class independent of view layer like it used to be,